### ISSUE TITLE
Consider a disclaimer or example for setting image sampler filter to Nearest to avoid sprite bleed

### ISSUE THREAD

**Issue by JoeHowarth** [Droggelbecher/bevy-fast-tilemap#27]
Consider adding an example that sets the image sampler to use nearest filtering to avoid sprite bleed. I didn't know linear filtering was set by default and it took quite a while to a) diagnose the issue and b) determine the best way to fix it. Along the way I tried padding which helped, but ultimately wasn't the real issue. 
This is addressed generally in https://github.com/bevyengine/bevy/discussions/1289, so if you feel a full example isn't warranted with boilerplate to actually set the value, consider at least linking to this issue?

With bleed:
<img width="506" alt="Screenshot 2024-05-05 at 8 32 04 PM" src="https://github.com/Droggelbecher/bevy-fast-tilemap/assets/22804465/9c1cf92a-fa55-4cf3-8194-351226f0f463">

After padding but still border issues
<img width="548" alt="Screenshot 2024-05-06 at 12 58 25 PM" src="https://github.com/Droggelbecher/bevy-fast-tilemap/assets/22804465/19655f86-6374-462c-9468-cb23f10a3927">
<img width="865" alt="Screenshot 2024-05-06 at 12 58 33 PM" src="https://github.com/Droggelbecher/bevy-fast-tilemap/assets/22804465/4a23982e-0989-4f41-a52b-f2acde2d81c0">

After setting filter to Nearest
<img width="514" alt="Screenshot 2024-05-06 at 1 33 42 PM" src="https://github.com/Droggelbecher/bevy-fast-tilemap/assets/22804465/bfe2c8a2-c68f-496b-9207-e5ab6c6be04b">

My code. Note this won't work out of the box because you don't have the required files on disk:
```rust
use bevy::{
    asset::{self, LoadState},
    diagnostic::{FrameTimeDiagnosticsPlugin, LogDiagnosticsPlugin},
    ecs::schedule::ScheduleLabel,
    math::{uvec2, vec2},
    prelude::*,
    render::{render_resource::FilterMode, texture::ImageSampler},
    window::PresentMode,
};
use bevy_fast_tilemap::{map::DefaultUserData, FastTileMapPlugin, Map, MapBundleManaged};

mod mouse_controls_camera;
use mouse_controls_camera::MouseControlsCameraPlugin;

#[derive(Resource)]
struct TileTerrainIndices(Vec<u32>);
#[derive(Resource)]
struct TextureAtlasHandle(Handle<Image>);

fn main() {
    App::new()
        .add_plugins((
            DefaultPlugins.set(WindowPlugin {
                primary_window: Some(Window {
                    title: String::from("Fast Tilemap example"),
                    resolution: (1820., 420.).into(),
                    // disable vsync so we can see the raw FPS speed
                    present_mode: PresentMode::Immediate,
                    ..default()
                }),
                ..default()
            }),
            LogDiagnosticsPlugin::default(),
            // FrameTimeDiagnosticsPlugin::default(),
            MouseControlsCameraPlugin::default(),
            FastTileMapPlugin::default(),
        ))
        .add_systems(Startup, startup)
        .add_systems(Update, ensure_texture_filter)
        // .add_systems(FixedUpdate, update_map)
        // Performance-wise you can step this much faster but it'd require an epillepsy warning.
        .insert_resource(Time::<Fixed>::from_seconds(0.2))
        .run();
}

fn ensure_texture_filter(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
    mut materials: ResMut<Assets<Map>>,
    mut images: ResMut<Assets<Image>>,
    mut done: Local<bool>,
    handle: Res<TextureAtlasHandle>,
    tile_terrain_indices: Res<TileTerrainIndices>,
) {
    if dbg!(*done) {
        return;
    }

    let handle = handle.0.clone();
    if dbg!(asset_server.load_state(&handle)) != LoadState::Loaded {
        return;
    }

    let atlas = images.get_mut(&handle).unwrap();
    atlas.sampler = ImageSampler::nearest();

    let map = Map::builder(uvec2(100, 100), handle, vec2(16., 16.))
        .with_padding(vec2(2., 2.), vec2(0., 0.), vec2(0., 0.))
        .build_and_initialize(|m| {
            // Initialize using a closure
            // Set all tiles in layer 0 to index 6
            for y in 0..m.size().y {
                for x in 0..m.size().y {
                    m.set(x, y, tile_terrain_indices.0[(x * 100 + y) as usize] - 1);
                }
            }
        });

    commands.spawn(MapBundleManaged {
        material: materials.add(map),
        ..default()
    });

    *done = true;
}

fn startup(mut commands: Commands, asset_server: Res<AssetServer>) {
    commands.spawn(Camera2dBundle::default());
    let handle = asset_server.load::<Image>("sprite_images/atlas-1.png");
    commands.insert_resource(TextureAtlasHandle(handle));

    let tiled_map_json = std::fs::read_to_string("./assets/SimpleMap.json").unwrap();
    let json = serde_json::from_str::<serde_json::Value>(&tiled_map_json).unwrap();
    let data = json
        .get("layers")
        .unwrap()
        .as_array()
        .unwrap()
        .get(0)
        .unwrap()
        .get("data")
        .unwrap()
        .as_array()
        .unwrap();
    let data = data
        .iter()
        .map(|v| serde_json::from_value::<u32>(v.clone()).unwrap())
        .collect::<Vec<_>>();
    commands.insert_resource(TileTerrainIndices(data));
}
```

**Comment by Droggelbecher** [comment]
We're doing something very similar automatically in https://github.com/Droggelbecher/bevy-fast-tilemap/blob/master/src/map.rs#L402 but it seems for some reason that does not (reliably) work, but it's completely unclear to me as to why currently.

**Comment by JoeHowarth** [comment]
Ahh I believe your code will work only if the map was built before the image asset was loaded, right?

In my app there's an asset loading state, then everything is initialized once it's been loaded already.

**Comment by Droggelbecher** [comment]
I think I fixed it in 0.7.4, would you care to give it a go for your usecase?
